src/pipecat/processors/frame_processor.py

```diff
@@ -28,68 +27,13 @@ class FrameDirection(Enum):
     DOWNSTREAM = 1
     UPSTREAM = 2
 
-
-class FrameProcessorMetrics:
-    def __init__(self, name: str):
-        self._name = name
-        self._start_ttfb_time = 0
-        self._start_processing_time = 0
-        self._should_report_ttfb = True
-
-    async def start_ttfb_metrics(self, report_only_initial_ttfb):
-        if self._should_report_ttfb:
-            self._start_ttfb_time = time.time()
-            self._should_report_ttfb = not report_only_initial_ttfb
-
-    async def stop_ttfb_metrics(self):
-        if self._start_ttfb_time == 0:
-            return None
-
-        value = time.time() - self._start_ttfb_time
-        logger.debug(f"{self._name} TTFB: {value}")
-        ttfb = {
-            "processor": self._name,
-            "value": value
-        }
-        self._start_ttfb_time = 0
-        return MetricsFrame(ttfb=[ttfb])
-
-    async def start_processing_metrics(self):
-        self._start_processing_time = time.time()
-
-    async def stop_processing_metrics(self):
-        if self._start_processing_time == 0:
-            return None
-
-        value = time.time() - self._start_processing_time
-        logger.debug(f"{self._name} processing time: {value}")
-        processing = {
-            "processor": self._name,
-            "value": value
-        }
-        self._start_processing_time = 0
-        return MetricsFrame(processing=[processing])
-
-    async def start_llm_usage_metrics(self, tokens: dict):
-        logger.debug(
-            f"{self._name} prompt tokens: {tokens['prompt_tokens']}, completion tokens: {tokens['completion_tokens']}")
-        return MetricsFrame(tokens=[tokens])
-
-    async def start_tts_usage_metrics(self, text: str):
-        characters = {
-            "processor": self._name,
-            "value": len(text),
-        }
-        logger.debug(f"{self._name} usage characters: {characters['value']}")
-        return MetricsFrame(characters=[characters])
-
-
 class FrameProcessor:
 
     def __init__(
             self,
             *,
             name: str | None = None,
+            metrics: FrameProcessorMetrics = FrameProcessorMetrics,
```

Comment:
I think this should be `FrameProcessorMetrics()` instead of passing a class. It's possible that other metrics implementation need additional arguments (api keys, user name, etc). If we hide the construction we would not be able to do that.

What we can do (since we need the name) is add a `FrameProcessorMetrics.set_processor_name()`:

```
self._metrics = metrics
metrics.set_processor_name(self.name)
```

Reply:
@aconchillo with `metrics: FrameProcessorMetrics = FrameProcessorMetrics()`, the` FrameProcessorMetrics()` instance is created once when the `FrameProcessor` class is defined. As a result, every time I create a new FrameProcessor, it uses the same instance of FrameProcessorMetrics, and it's a problem. My idea was to use `metrics: FrameProcessorMetrics = None` and 
```python
self._metrics = metrics or FrameProcessorMetrics() 
self._metrics.set_processor_name(self.name)
``` 
 to create a new instance of `FrameProcessorMetrics` inside the constructor if metrics is not provided. Let me know if i'm missing something, thanks.

Reply:
Oh yes! You are totally right :sweat_smile:  . Well, the important point was for people to be able to create instances from outside which we now can do! :smile: